rpcserver.go

```diff
@@ -1878,19 +1878,22 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 	localFundingAmt := btcutil.Amount(in.LocalFundingAmount)
 	remoteInitialBalance := btcutil.Amount(in.PushSat)
 
-	// If the FundMax flag is set, ensure that the
-	// acceptable minimum local amount adheres to the
-	// amount to be pushed to the remote, and to current
-	// rules.
+	// If the FundMax flag is set, ensure that the acceptable minimum local
+	// amount adheres to the amount to be pushed to the remote, and to
+	// current rules, while also respecting the settings for the maximum
+	// channel size.
+	var fundUpToMaxAmt btcutil.Amount
 	if in.FundMax {
-		// Use as minimum local amount the larger of
-		// the amount to be pushed to the remote or
-		// the allowed minimum funding size.
-		if remoteInitialBalance >= funding.MinChanFundingSize {
+		// As default value for the minimum local amount use the larger
+		// of the amount to be pushed to the remote or the allowed
+		// minimum channel size.
```

Comment:
To be honest, these comments don't really help me to understand _why_ something is done. They just describe the code below in words. We should add the context from the discussions in this PR instead. Otherwise someone looking at this in a year or so won't know what was discussed.

Reply:
Tried to make it clearer here: https://github.com/lightningnetwork/lnd/pull/6903/commits/58bc9bdc6e9a0d335c92585665e02d8fad8c653b#diff-674c79ba02b6e9c45a994388392689814f3538800351b2fd43fb3ade21e1effbR1916